src/System.Buffers.Primitives/System/Buffers/OwnedBuffer.cs

```diff
@@ -13,45 +12,17 @@ public abstract class OwnedBuffer<T> : IDisposable, IRetainable
 
         public abstract int Length { get; }
 
-        public abstract Span<T> Span { get; }
+        public abstract Span<T> AsSpan(int index, int length);
```

Comment:
>  Span.Slice(0, 10) would first have to read the whole file and then create a span of just first ten bytes.

I don't completely understand this. The cost of creating a span isn't proportional to the input buffer size it is wrapping, correct? So why does Span.Slice(0, 10) read the whole file whereas AsSpan(0, 10) does not?